### PR TITLE
:book: Refactor the documentation for compatibility with the docs.scs.community

### PR DESCRIPTION
**Pull request by matofeder**
related to https://github.com/SovereignCloudStack/docs/issues/136

### CODE REVIEW

**Review by michal-gubricky — APPROVED** [review]
lgtm

File: docs/quickstart.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+Install the [envsubst] Go package. It is required to enable the expansion of variables specified in CSPO and CSO manifests.
+
+```bash
+GOBIN=/tmp go install github.com/drone/envsubst/v2/cmd/envsubst@latest
+```
+
+Get the latest CSO release version and apply CSO manifests to the management cluster.
+
+```bash
```
Comment (chess-knight) [inline docs/quickstart.md:73]: The latest can be retrieved by `https://github.com/SovereignCloudStack/cluster-stack-operator/releases/latest/download/cso-infrastructure-components.yaml`

We do not need `jq`(missing from prerequisites) then

File: docs/quickstart.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+To enable communication between the CSPO and the Cluster API Provider for OpenStack (CAPO) with the OpenStack API, it is necessary to generate a secret containing the access data (clouds.yaml).
+Ensure that this secret is located in the identical namespace as the other Custom Resources.
+
+> [!NOTE]  
+> The default value of `cloudName` is configured as `openstack`. This setting can be overridden by including the `cloudName` key in the secret. Also, be aware that the name of the secret is expected to be `openstack` unless it is not set differently in OpenStackClusterStackReleaseTemplate in `identityRef.name` field.
+
+```bash
+kubectl create secret generic openstack --from-file=clouds.yaml=path/to/clouds.yaml
+
+# Patch the created secrets so they are automatically moved to the target cluster later.
```
Comment (chess-knight) [inline docs/quickstart.md:43]: We can use `kubectl label` instead:
`kubectl label secret openstack clusterctl.cluster.x-k8s.io/move=`

File: docs/quickstart.md [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+```
+
+Proceed to apply the `ClusterStack` to the management cluster. For more details, refer to `examples/clusterstack.yaml`:
+
+```bash
+kubectl apply -f <path-to-openstack-clusterstack>
+```
+
+Please be patient and wait for the operator to execute the necessary tasks.
+If your `ClusterStack` object encounters no errors and `openstacknodeimagereleases` is ready, you can deploy a workload cluster.
```
Comment (chess-knight) [inline docs/quickstart.md:109]: IMO this is only partially true. There are other objects openstackclusterstackrelease and clusterstackrelease, which can be ready. ClusterStackRelease needs to be ready and ClusterStack has to report usable versions.

File: docs/controllers.md [hunk 4]
```diff
@@ -9,3 +11,14 @@ The OpenStackClusterStackRelease controller’s main responsibilities are:
 - Update the OpenStackClusterStackRelease status to `ready` once all related OpenStackNodeImageReleases are also `ready`
 
 ![OSCSR controller](./images/openstackclusterstackrelease-controller.png "OSCSR controller")
```
Comment (chess-knight) [inline docs/controllers.md:13]: I suggest modifying this img according to https://github.com/SovereignCloudStack/cluster-stack-provider-openstack/pull/92

File: docs/quickstart.md [hunk 5]
```diff
@@ -0,0 +1,139 @@
+# Quickstart
+
+This section guides you through all the necessary steps to create a workload Kubernetes cluster on top of the OpenStack infrastructure. The guide describes a path that utilizes the [clusterctl] CLI tool to manage the lifecycle of a CAPI management cluster and employs [kind] to create a local non-production management cluster.
```
Comment (chess-knight) [inline docs/quickstart.md:3]: I suggest adding a link for `CAPI`

File: docs/quickstart.md [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+CSPO_VERSION=$(curl https://api.github.com/repos/SovereignCloudStack/cluster-stack-provider-openstack/releases/latest -s | jq .name -r)
+# Apply CSPO manifests
+curl -sSL https://github.com/SovereignCloudStack/cluster-stack-provider-openstack/releases/download/${CSPO_VERSION}/cspo-infrastructure-components.yaml | /tmp/envsubst | kubectl apply -f -
+```
+
+## Create the workload cluster
+
+To transfer the credentials stored in the mentioned secret [above](#create-a-secret-for-openstack-access) to the operator,
+create an `OpenStackClusterStackReleaseTemplate` object and specify this secret in the `identityRef` field.
+The `clouds.yaml` file may contain one or more clouds, so users must specify the desired connection to a specific cloud using the `cloudName` field.
```
Comment (chess-knight) [inline docs/quickstart.md:93]: This `cloudName` should be inside Secret or oscsrt ?

File: docs/quickstart.md (hunk 1 above)
Comment (matofeder) [reply docs/quickstart.md:73]: good point, but let's prefer explicit before implicit, and I would rather prefer to add `jq` to the prerequisites

Comment (chess-knight) [reply docs/quickstart.md:73]: Ok, I agree

File: docs/quickstart.md (hunk 6 above)
Comment (matofeder) [reply docs/quickstart.md:93]: true, I removed this section as the secret creation and proper explanation are linked here `[above](#create-a-secret-for-openstack-access)`

Comment (matofeder) [reply docs/quickstart.md]: fixed: c380e5fa26ad5d4725148dfaf08bed0f842afcc8

File: docs/quickstart.md (hunk 5 above)
Comment (matofeder) [reply docs/quickstart.md]: fixed: c380e5fa26ad5d4725148dfaf08bed0f842afcc8

File: docs/controllers.md (hunk 4 above)
Comment (matofeder) [reply docs/controllers.md:13]: fixed: c380e5fa26ad5d4725148dfaf08bed0f842afcc8

File: docs/quickstart.md (hunk 3 above)
Comment (matofeder) [reply docs/quickstart.md]: fixed: c380e5fa26ad5d4725148dfaf08bed0f842afcc8

File: docs/quickstart.md (hunk 2 above)
Comment (matofeder) [reply docs/quickstart.md]: fixed: c380e5fa26ad5d4725148dfaf08bed0f842afcc8

File: docs/quickstart.md (hunk 1 above)
Comment (matofeder) [reply docs/quickstart.md:74]: fixed: c380e5fa26ad5d4725148dfaf08bed0f842afcc8

File: README.md [hunk 7]
```diff
@@ -4,173 +4,38 @@
 [![Go Report Card](https://goreportcard.com/badge/github.com/sovereignCloudStack/cluster-stack-provider-openstack)](https://goreportcard.com/report/github.com/sovereignCloudStack/cluster-stack-provider-openstack)
```
Comment (chess-knight) [inline README.md:4]:
```suggestion
[![Go Report Card](https://goreportcard.com/badge/github.com/SovereignCloudStack/cluster-stack-provider-openstack)](https://goreportcard.com/report/github.com/SovereignCloudStack/cluster-stack-provider-openstack)
```

File: docs/quickstart.md [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+GOBIN=/tmp go install github.com/drone/envsubst/v2/cmd/envsubst@latest
+```
+
+Get the latest CSO release version and apply CSO manifests to the management cluster.
+
+```bash
+# Get the latest CSO release version
+CSO_VERSION=$(curl https://api.github.com/repos/SovereignCloudStack/cluster-stack-operator/releases/latest -s | jq .name -r)
+# Apply CSO manifests
+curl -sSL https://github.com/sovereignCloudStack/cluster-stack-operator/releases/download/${CSO_VERSION}/cso-infrastructure-components.yaml | /tmp/envsubst | kubectl apply -f -
```
Comment (chess-knight) [inline docs/quickstart.md:78]:
```suggestion
curl -sSL https://github.com/SovereignCloudStack/cluster-stack-operator/releases/download/${CSO_VERSION}/cso-infrastructure-components.yaml | /tmp/envsubst | kubectl apply -f -
```

**Review by chess-knight — APPROVED** [review]

**Pull request merged**